backend/app/Services/Customer/CustomerMetricsService.php

```
@@ -0,0 +1,53 @@
+class CustomerMetricsService
+{
+    public function recordCompletion(WorkOrder $workOrder): void
+    {
+        if ($workOrder->customer_totals_counted || $workOrder->customer_id === null) {
+            return;
+        }
+
+        $revenue = (float) $workOrder->produced_qty * (float) ($workOrder->unit_price ?? 0);
+
+        DB::transaction(function () use ($workOrder, $revenue): void {
+            $customer = Customer::whereKey($workOrder->customer_id)->lockForUpdate()->first();
```

PHPStan (2.2.2)

[error] Call to an undefined static method App\Models\Customer::whereKey().


(staticMethod.notFound)